src/libstd/io/mod.rs

```diff
@@ -322,6 +576,17 @@ pub trait Write {
     ///
     /// It is **not** considered an error if the entire buffer could not be
     /// written to this writer.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::io::Write;
+    ///
+    /// let mut buffer = Vec::new();
+    ///
+    /// buffer.write(b"some bytes").unwrap();
```

Comment:
This may want to bind the return value of `write` to emphasize that this is **not** writing all of the bytes. 

Ah and also the same comment as below about how this may not want a flush afterwards